health-sdk/sdk/src/main/java/net/gini/android/health/sdk/review/ReviewFragment.kt

```diff
@@ -574,6 +589,10 @@ class ReviewFragment private constructor(
             }
             ReviewViewModel.PaymentNextStep.ShowOpenWithSheet -> viewModel.paymentProviderApp.value?.let { showOpenWithDialog(it) }
             ReviewViewModel.PaymentNextStep.ShowInstallApp -> viewModel.paymentProviderApp.value?.let { showInstallAppDialog(it) }
+            is ReviewViewModel.PaymentNextStep.OpenSharePdf -> {
+                binding.loading.isVisible = false
```

Comment:
It's still needed here, as the visibility isn't set to GONE from the viewModel. I opted for this in case there is any extra functionality that we might want to add in the UI, so the UI layer decides when to hide the loading indicator.

I added the `setLoadingVisibility` step in the VM because it's the VM which decides what the next step should be, based on all the variables that we monitor. It currently sends a `boolean` parameter to the `sharedFlow` but it's always `true` at the moment. I opted for this for the same reason as I mentioned above, if there's any case in which we should hide the loading from the VM at any point in the future (so we don't need to refactor). If you consider this to be an overworking of a simple thing, please let me know and I can switch it over to a plain `object`.